python/ql/src/experimental/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -357,45 +357,47 @@ private module ArgumentPassing {
   /**
    * A type representing a mapping from argument indices to parameter indices.
    * We currently use two mappings: NoShift, the identity, used for ordinary
-   * function calls, and ShiftOne which is used for calls where an extra argument
+   * function calls, and ShiftOneUp which is used for calls where an extra argument
    * is inserted. These include method calls, constructor calls and class calls.
    * In these calls, the argument at index `n` is mapped to the parameter at position `n+1`.
    */
   newtype TArgParamMapping =
     TNoShift() or
-    TShiftOne()
+    TShiftOneUp()
 
   /** A mapping used for parameter passing. */
   abstract class ArgParamMapping extends TArgParamMapping {
-    bindingset[result]
-    abstract int getArgN(int paramN);
+    /** Gets the index of the parameter that corresponds to the argument at index `argN`. */
+    bindingset[argN]
+    abstract int getParamN(int argN);
 
-    string toString() { none() }
+    /** Gets a textual representation of this element. */
+    string toString() { result = "ArgParamMapping" }
```

Comment:
Ah, actually I was thinking this would make more sense to have on `NoShift` and `ShiftOneUp` so we can tell them apart. :slightly_smiling_face: Just make the base one `abstract` and `override` as needed in the subclasses.

Reply:
Done